addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -263,6 +259,156 @@ class DataEgressService extends Service {
+    const curVersion = parseInt(egressStoreInfo.ver, 10);
+    const key = `${egressStoreInfo.id}/${egressStoreInfo.egressStoreName}-ver${curVersion}.json`;
+    try {
+      const objectList = await s3Service.listAllObjects({
+        Bucket: egressStoreInfo.s3BucketName,
+        Prefix: egressStoreInfo.s3BucketPath,
+      });
+      await Promise.all(
+        _.map(objectList, async obj => {
+          const latestVersion = await s3Service.getLatestObjectVersion({
+            Bucket: egressStoreInfo.s3BucketName,
+            Prefix: obj.Key,
+          });
+          obj.VersionId = latestVersion.VersionId;
+          obj.Owner = latestVersion.Owner;
+          return obj;
+        }),
+      );
+      const params = {
+        Bucket: egressNotificationBucketName,
+        Key: key,
+        Body: JSON.stringify({ objects: objectList }),
+        ContentType: 'application/json',
+      };
+      await s3Service.putObject(params);
+    } catch (error) {
+      throw this.boom.badRequest(
+        `Error in preparing EgressStoreSnapshot, bucket:${egressNotificationBucketName}, key: ${key}`,
+        true,
+      );
+    }
+    return { bucket: egressNotificationBucketName, key };
+  }
+
+  async notifySNS(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    const curUser = _.get(requestContext, 'principalIdentifier.uid');
+    if (!enableEgressStore || enableEgressStore.toUpperCase() !== 'TRUE') {
+      throw this.boom.forbidden('Unable to create Egress store since this feature is disabled', true);
+    }
+
+    const egressStoreInfo = await this.getEgressStoreInfo(environmentId);
+    if (!egressStoreInfo.isAbleToSubmitEgressRequest) {
+      throw this.boom.badRequest(
+        `Egress Store:${egressStoreInfo.id} is not ready for egress. Please contact your administrator for more information.`,
+        true,
+      );
+    }
+    const isEgressStoreOwner = egressStoreInfo.createdBy === curUser;
+    if (!isAdmin(requestContext) && !isEgressStoreOwner) {
+      throw this.boom.forbidden(
+        `You are not authorized to submit egress request. Please contact your administrator for more information.`,
+        true,
+      );
+    }
+    const egressStoreObjectList = await this.prepareEgressStoreSnapshot(egressStoreInfo);
+
+    // update dynamodb info
+    const egressStoreDdbLockId = `egress-store-ddb-access-${egressStoreInfo.id}`;
+    if (egressStoreInfo.status.toUpperCase() !== PENDING_STATUS_CODE) {
+      egressStoreInfo.status = PENDING_STATUS_CODE;
+    }
+    egressStoreInfo.updatedBy = curUser;
+    egressStoreInfo.updatedAt = new Date().toISOString();
+    egressStoreInfo.isAbleToSubmitEgressRequest = false;
+    egressStoreInfo.egressStoreObjectListLocation = `arn:aws:s3:::${egressStoreObjectList.bucket}/${egressStoreObjectList.key}`;
+    const curVersion = parseInt(egressStoreInfo.ver, 10);
```

Comment:
Add a comment in the code explaining what `10` represents.

Reply:
fixed in commit: 18639d639b979c9054fa5d24fa94b3a0bdd98f11